docs/source/internals/data_storage.rst

```diff
@@ -1,32 +1,104 @@
-.. todo::
+.. _internal_architecture:data-storage:
 
-    .. _internal_architecture:data-storage:
+************
+Data storage
+************
 
-    ************
-    Data storage
-    ************
+.. todo::
 
     .. _internal_architecture:orm:database:
 
-    Database schema & migrations
-    ============================
+    title: Database schema & migrations
 
     `#4035`_
 
     .. _internal_architecture:orm:repository:
 
-    File repository
-    ===============
+    title: File repository
 
     `#4036`_
 
-    .. _internal_architecture:orm:export:
+.. _internal_architecture:orm:archive:
+
+AiiDA archive format
+====================
+
+An AiiDA archive file is a ``.zip`` or ``.tar.gz`` file with the following content:
+
+* ``metadata.json`` file containing information on the version of AiiDA as well as the database schema.
+* ``data.json`` file containing the nodes and their links.
+* ``nodes/`` directory containing the repository files corresponding to the nodes.
+
+.. _metadata-json:
+
+``metadata.json``
+-----------------
+
+This file contains important information, and it is necessary for the correct interpretation of ``data.json``.
+Apart from the data schema, the AiiDA and archive versions are also mentioned.
+This is used to avoid any incompatibilities among different versions of AiiDA.
+It should be noted that the schema described in ``metadata.json`` is related to the data itself - abstracted schema focused on the extracted information - and not how the data is stored in the database (database schema).
+This makes the import/export mechanism transparent to the database system used, backend selected and how the data is organised in the database (database schema).
+
+Let's have a look at the contents of ``metadata.json``:
+
+.. literalinclude :: includes/metadata.json
+   :language: json
+
+At the beginning of the file, we see the version of the archive file (under `export_version`) and the version of the AiiDA code.
```

Comment:
```suggestion
At the beginning of the file, we see the version of the archive file (under ``export_version``) and the version of the AiiDA code.
```

Reply:
I think here and below sometimes single backpacks are used, sometimes stars etc - I suggest using double backticks everywhere we refer to a literal string (e.g. the key of one of the JSON files) - I didn't put a comment on each of them